sdk/core/azure-core/src/context.cpp

```diff
@@ -5,7 +5,7 @@
 
 using namespace Azure::Core;
 
-const Context Context::ApplicationContext;
+Context Context::ApplicationContext;
```

Comment:
Now that this isn't const any code including a dependent package I include can replace the application context by doing.
   `Context::ApplicationContext = Context::CreateNewRoot();`

How do I know the application context is the application context I configured?

Reply:
That's the way the code worked until February 2023 when I incorrectly made ApplicationContext `const`. The design of ApplicationContext was never that it should be `const` - I made this change by accident as an attempt to enforce the idea that all contexts were immutable (this part of the change was overkill from removing a design element that allowed developers to set a single tracing provider for the entire process).

So until February 2023, any developer could have changed ApplicationContext.